compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -779,7 +777,7 @@ object ProtoTypes {
         pt => bounds :: Nil,
         pt => represents.orElse(defn.AnyType))
     constrained(poly, untpd.EmptyTree, alwaysAddTypeVars = true, nestingLevel)
-      ._2.head.tpe.asInstanceOf[TypeVar]
+      ._2.head
 
```

Comment:
From what I can see, the changes are only about APIs, which simplifies usage in several places, the logic is the same. /cc: @odersky